native_shortuuid/admin.py

```diff
@@ -1,34 +1,35 @@
+import django.conf
 import django.db.models
 
 import native_shortuuid
 
 
 class NativeUUIDSearchMixin:
-    def is_valid_shortuuid(self, seach_term):
+    search_uuid_fields = []
+
+    def is_valid_shortuuid(self, search_term):
         try:
             # It's not decode-able, thus, not a shortuuid
-            native_shortuuid.decode(seach_term)
+            native_shortuuid.decode(search_term)
         except ValueError:
             return False
 
         return True
 
     def get_search_fields(self, request):
         search_fields = super().get_search_fields(request)
-        if not hasattr(self, '_uuid_search_fields'):
-            self._uuid_search_fields = tuple(x for x in search_fields if x.endswith('uuid'))
-            self._non_uuid_search_fields = tuple(x for x in search_fields if not x.endswith('uuid'))
-        return self._non_uuid_search_fields or self._non_uuid_search_fields + ('id', )
+        if getattr(django.conf.settings, 'ADMIN_EXTRACT_UUID_SEARCH_FIELDS', False):
+            self.search_uuid_fields = tuple(x for x in search_fields if x.endswith('uuid'))
+            search_fields = tuple(x for x in search_fields if not x.endswith('uuid'))
```

Comment:
@laithfawwaz if there is only uuid fields in the list the search_fields will be empty and this will cause an error
if there it no search_fields return the id field at least

Reply:
@hadayda oh, got you, will fix this